### ISSUE TITLE
Persisting Kernel across page reloads

### ISSUE THREAD

**Issue by BrandonEscamilla** [datalayer/jupyter-ui#92]
Hi everyone,

Is there a way to maintain the same kernel even after page reloads? I'd like to preserve the kernel and avoid it being reset. Any suggestions on achieving this? Your help is greatly appreciated. Thank you!

P.S.  I've been considering using ﻿redux-persist since I noticed that the kernel is a part of the Redux state. Unfortunately, I haven't been successful in getting it to work so far.

**Comment by BrandonEscamilla** [comment]
Hey @echarles, I'm using the latest release of Jupyter UI (version 0.6.5). Initially, I thought it might be related to my Redux setup, so I removed all the Redux configuration, but the problem still persists. As a result, my Jupyter server is getting overwhelmed with numerous unused kernels. Do you have any ideas or suggestions on how I can address this issue?

**Comment by BrandonEscamilla** [comment]
I created a new application from scratch to verify if the issue was with my setup. However, I encountered the same behavior. It's worth considering that I may be overlooking something or have a misunderstanding of the kernel's definition. Nonetheless, this is the method I use to determine if the kernel is created or reused. Thanks in advance! 

```jsx
"use client";
import { Jupyter, Notebook } from '@datalayer/jupyter-react'

export default function Home() {
  const NOTEBOOK_UID = 'notebook-uid'
  return (
    <main className="w-full h-screen">      
      <Jupyter
        jupyterServerHttpUrl=""
        jupyterServerWsUrl=""
        jupyterToken=""                    
      >
        <Notebook
          path="/Untitled.ipynb"
          height='calc(100vh - 2.6rem)'
          cellSidebarMargin={120}                
          uid={NOTEBOOK_UID}                
        />
      </Jupyter>
    </main>
  )
}

```
Video url: https://share.cleanshot.com/VRq7Dndp

**Comment by echarles** [comment]
Thx for very useful video. A "default kernel" is always started on page refresh, but the kernel to which the notebook connects is managed by the notebook context, so you should be connected to the same kernel, whatever it is, on page refresh (you can check that your variables are still there).

Anyway, this is suboptimal, so I am refactoring all that to solve this and hopefully other issues you reported.

Stay tuned!

**Comment by BrandonEscamilla** [comment]
Good to know! Currently, I'm unmounting the kernel whenever the user reloads or closes the page to prevent overwhelming the server with multiple kernels. However, this approach does result in the deletion of the variables you mentioned.

**Comment by echarles** [comment]
I have released 0.7.0 which fixed this. Can you try on your env and close this if it work for you? Thx again!

**Comment by BrandonEscamilla** [comment]
Hi @echarles, it appears that the issue has been resolved. I tested it and now it's only starting one kernel, and keeping the same after page reloads. Thank you for your assistance with this! I will close the issue.
